File: 70-climbing-stairs/70-climbing-stairs.py
class Solution(object):
    
    dp = [1, 2]
    
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n <= len(self.dp):
            return self.dp[n - 1]
        result = self.climbStairs(n - 1) + self.climbStairs(n - 2)
        for i in range(len(self.dp), n, 1):
            self.dp.append(0)
        self.dp[n - 1] = result
        return self.dp[n-1]
        
        
        

File: 70-climbing-stairs/70-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if (n == 1) {
            return 1;
        }
        if (n == 2) {
            return 2;
        }
        int pp = 1;
        int p = 2;
        int result = 2;
        for (int i = 3; i <= n; i++) {
            result = p + pp;
            pp = p;
            p = result;
        }
        return result;
    }
}